server/createSVG.js

```diff
@@ -58,29 +64,103 @@ var Tools = {
 		return renderPath(el, pathstring);
 	},
 	"Straight line": function (el) {
-		var pathstring = "M" + el.x + " " + el.y + "L" + el.x2 + " " + el.y2;
+		const pathstring = "M" + el.x + " " + el.y + "L" + el.x2 + " " + el.y2;
 		return renderPath(el, pathstring);
 	}
 };
 
+exportTools = {
+	"Pencil": function (el) {
+		if (!el._children) return "";
+		let pathstring;
+		switch (el._children.length) {
+			case 0:
+				return "";
+			case 1:
+				pathstring = "M" + el._children[0].x + " " + el._children[0].y +
+					"L" + el._children[0].x + " " + el._children[0].y;
+				break;
+			case 2:
+				pathstring = "M" + el._children[0].x + " " + el._children[0].y +
+					"C" + el._children[0].x + " " + el._children[0].y + " " +
+					el._children[1].x + " " + el._children[1].y + " " +
+					el._children[1].x + " " + el._children[1].y;
+				break;
+			default:
+				pathstring = "M" + el._children[0].x + " " + el._children[0].y;
+				const pts = [
+					{type: "M", values: [el._children[0].x, el._children[0].y]},
+					{type: "C", values: [el._children[0].x, el._children[0].y,
+							el._children[1].x, el._children[1].y,
+							el._children[1].x, el._children[1].y]},
+				];
+				for (let i = 2; i < el._children.length; i++) {
+					//We add the new point, and smoothen the line
+					const ANGULARITY = 3; //The lower this number, the smoother the line
+					const prev_values = pts[pts.length - 1].values; // Previous point
+					const ante_values = pts[pts.length - 2].values; // Point before the previous one
+					const prev_x = prev_values[prev_values.length - 2];
+					const prev_y = prev_values[prev_values.length - 1];
+					const ante_x = ante_values[ante_values.length - 2];
+					const ante_y = ante_values[ante_values.length - 1];
+					const x = el._children[i].x;
+					const y = el._children[i].y;
+
+					//We don't want to add the same point twice consecutively
+					if ((prev_x === x && prev_y === y)
+						|| (ante_x === x && ante_y === y)) continue;
+
+					let vectx = x - ante_x,
+						vecty = y - ante_y;
+					const norm = Math.hypot(vectx, vecty);
+					const dist1 = dist(ante_x, ante_y, prev_x, prev_y) / norm,
+						dist2 = dist(x, y, prev_x, prev_y) / norm;
+					vectx /= ANGULARITY;
+					vecty /= ANGULARITY;
+					//Create 2 control points around the last point
+					const cx1 = prev_x - dist1 * vectx,
+						cy1 = prev_y - dist1 * vecty, //First control point
+						cx2 = prev_x + dist2 * vectx,
+						cy2 = prev_y + dist2 * vecty; //Second control point
+					prev_values[2] = cx1;
+					prev_values[3] = cy1;
+
+					npoint = {
+						type: "C", values: [
+							cx2, cy2,
+							x, y,
+							x, y,
+						]
+					};
+					pts.push(npoint);
+				}
+				for (let i = 1; i < pts.length; i++) {
+					pathstring += " " + pts[i].type + " " + pts[i].values.join(" ");
+				}
+		}
+
+		return renderPath(el, pathstring);
```

Comment:
Can't we refactor all this code in order to use the same code on the client and the server ?

Reply:
How would you go about accessing the function? We can't use `require` on the client side but we also don't want to pollute our namespace on the client side either.